src/world/map.c

```diff
@@ -1709,10 +1970,258 @@ static money32 map_set_land_height(int flags, int x, int y, int height, int styl
+		ebx = mapElement->clearance_height;
+		ebx -= *pdl;
+		if(ebx>=0)
+		{
+			ebx >>= 1;
+			if(ebx>eax)
+			{
+				//rct2: 0x663B09
+				RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = STR_SUPPORTS_CANT_BE_EXTENDED;
+				return MONEY32_UNDEFINED;
+			}
+		}
+	}
+loc_663B1A:
+	mapElement++;
+	if((mapElement-1)->flags&MAP_ELEMENT_FLAG_LAST_TILE)
+		goto loc_663AC4;
+	ecx = saved_ecx;
+	ebx = saved_ebx;
+	eax = saved_eax;
+	saved_ecx = ecx;
+	
+	ebp *= 0x260;
+	esi = (int)mapElement;
+	//callcode_push1(0x663B27, saved_ecx, &eax, &ebx, &ecx, (int *)&edx, (int *)&esi, &edi, &ebp); return ebx;
+	
+	/*
+	esi = ecx&0xFFFF;
+	esi <<= 8;
+	esi |= eax&0xFFFF;
+	esi >>= 3;
+	esi = RCT2_ADDRESS(RCT2_ADDRESS_TILE_MAP_ELEMENT_POINTERS, uint32)[esi/4];
+	*/
+	mapElement = map_get_first_element_at(eax/32, ecx/32); //esi
+	esi = (int)mapElement;
+	//callcode_push1(0x663B39, saved_ecx, &eax, &ebx, &ecx, (int *)&edx, (int *)&esi, &edi, &ebp); return ebx;
+	
+	while(mapElement->type&MAP_ELEMENT_TYPE_MASK)
+		mapElement++;
+	if(mapElement->type&0x40)
+	{
+		//ecx = *((uint8 *)esi+5)&0x1F; //Don't know which union member to use yet.
+		ecx = mapElement->properties.scenery.age&0x1F; //Don't know if this is the correct union member.
+		if(ecx!=0)
+		{
+			ecx = ecx*2-2;
+			*pbh = *pdl;
+			if((*pdh&0x1F))
+			{
+				*pbh += 2;
+				if((*pdh&0x10))
+				{
+					*pbh += 2;
+				}
+			}
+		loc_663B6A:
+			if(*pbh>ecx)
+			{
+				//rct2: 0x663C5A
+				mapElement++;
+				map_obstruction_set_error_text(mapElement);
+				return MONEY32_UNDEFINED;
+			}
+		}
+	}
+loc_663B72:
+	ecx = saved_ecx;
+	
+	esi = (int)mapElement;
+	//RCT2_CALLFUNC_X(0x663B73, &eax, &ebx, &ecx, (int *)&edx, (int *)&esi, &edi, &ebp); return ebx;
+
+	*pbh = *pdl;
+	if((*pdh&0xF))
+	{
+		*pbh += 2;
+		if((*pdh&0x10))
+		{
+			*pbh += 2;
+		}
+	}
+loc_663B85:
+	saved_ebx = ebx;
+	saved_edx = edx;
+	int saved_ebp = ebp;
+	
+	//callcode_push3(0x663B88, saved_ebx, saved_edx, saved_ebp, &eax, &ebx, &ecx, (int *)&edx, (int *)&esi, &edi, &ebp); return ebx;
+	
+	*pbl = *pdh;
+	*pdh = *pbh;
+	*pbl = 0xF;
+	ebp = (int)RCT2_ADDRESS(0x663CB9, void);
+	if(map_can_construct_with_clear_at(eax&0xFFFF, ecx&0xFFFF, *pdl, *pdh, (void *)ebp, *pbl)==false)
+		return MONEY32_UNDEFINED;
+	ebp = saved_ebp;
+	edx = saved_edx;
+	ebx = saved_ebx;
+	saved_eax = eax;
+	saved_ecx = ecx;
+	
+	//callcode_push2(0x663BA3, saved_eax, saved_ecx, &eax, &ebx, &ecx, (int *)&edx, (int *)&esi, &edi, &ebp); return ebx;
+	
+	/*
+	edi = ecx&0xFFFF;
+	edi <<= 8;
+	edi = (edi|eax)&0xFFFF;
+	edi >>= 3;
+	edi = (uint32)RCT2_ADDRESS(RCT2_ADDRESS_TILE_MAP_ELEMENT_POINTERS, rct_map_element *)[edi/4];
+	*/
+	edi = (int)map_get_first_element_at(eax/32, ecx/32);
+loc_663BB5:
+	//callcode_push2(0x663BB5, saved_eax, saved_ecx, &eax, &ebx, &ecx, (int *)&edx, (int *)&esi, &edi, &ebp); return ebx;
+	eax = ((rct_map_element *)edi)->type&MAP_ELEMENT_TYPE_MASK;
+	if(eax==0x14 || eax==0xC || ((rct_map_element *)edi)->flags&0x10 || edi==esi)
```

Comment:
See above change to a do while loop 0x14 and 0xC will be MAP_ELEMENT_TYPE don't have the time to look up which ones.